### ISSUE TITLE
Fork Raspi-UA-Netinstall and make Pi-Hole The Distribution!

### ISSUE THREAD

**Issue by dschaper** [pi-hole/pi-hole#76]
Long term goal of creating a Pi-Hole Distro for user one-shot install from blank SD-Card to full ad-blocker (in less time than it takes to have a pizza delivered.)

**Comment by meliton** [comment]
I was also thinking about what happens when you take your Pi-hole and set it up elsewhere. I'm thinking along the lines of any off-the-shelf network appliance (i.e. SOHO router) that you plug into your network, run the installation CD, and then configure it via a web interface. In this context, the Pi-hole can DHCP into the network, then the installation CD can be used to find the Pi-hole and then fire up the web interface to finalize the configuration.

**Comment by dschaper** [comment]
There is a web front end to debconf, but I haven't used it and from what I am reading, it's not all that great, just a proof of concept. I know OpenWRT has been ported to the RasPi, but Pi's are good for DNS/DHCP yet they make craptastics routers. (The Ethernet port is run over the USB, so you're constrained in your bandwidth capabilities, especially if you add a usb Ethernet dongle to it. I've read a little about doing routing over the GPIO, but I don't know if the hardware is readily available.)

**Comment by dschaper** [comment]
Just want to ping @Fourdee on this. I see that his distro has install capabilities for PiHole, and maybe there is something we could work on, or get some guidance on with regards to this goal.

**Comment by Fourdee** [comment]
@dschaper Here is the current DietPi/Pihole automated installation image: http://dietpi.com/downloads/images/Pi-hole_RPi.zip
The user just needs to login to start setup and input static IP at the prompt.

For the next DietPi release (v104), we are supporting fully automated installations. This will allow for a full automatic and optimized software installation (eg: pihole), without any user input or login.

If you need any specific info/sourcecode for the PiHole install, just let me know.

**Comment by Fourdee** [comment]
I've released DietPi-Automation under BETA if you guys want to try it. This will automatically install, configure and optimize Pihole + Web stats with no user input on a fresh image:
http://fuzon.co.uk/phpbb/viewtopic.php?f=8&t=273&p=1140#p1140

**Comment by dschaper** [comment]
Thanks! I'll take a look at it this afternoon and see...

**Comment by jacobsalmela** [comment]
Users seem to enjoy DietPi, but I haven't been able to update it there.

**Comment by dschaper** [comment]
Okay, I tried the Automated Install, and it didn't look like it installed. So I did a manual install. It looks like it's installing Apache instead of lighttpd, and the version of gravity.sh is an old one. I couldn't find the other support scripts either, and I couldn't get to the admin page at 192.168.0.100/admin/admin.php. Is there a way to update the scripts, Pi-Hole is under very active development and I'm trying to get a .deb of it going, but that .deb will be dependent upon our choice of web server, it really doesn't need an entire LASP stack? Maybe we can collaborate on a Pi-Hole specific distribution that more closely ties to how Pi-Hole is being developed to envision?

**Comment by jacobsalmela** [comment]
I spoke with @Fourdee a while back and he told me there was a way to keep Pi-hole updated.  It needed some work on my end.  Also, yes, he does use Apache instead of lighttpd, so I'm not sure how well everything will tie together.  It would be great if we could somehow keep DietPi and Pi-hole separate, so we can each work on our projects independently.

**Comment by dschaper** [comment]
Well, let me know... I did find the web interface and it seems to be a little broken, I don't know how Apache is installed when the basic-install.sh script pulls in lighttpd... Some things to think about...

**Comment by Fourdee** [comment]
```
Okay, I tried the Automated Install, and it didn't look like it installed.
```
I just tried it using the example PiHole automation file. Working fine.

```
 I couldn't get to the admin page at 192.168.0.100/admin/admin.php
```
http://192.168.0.100/pihole
When the PiHole image was created, the web interface stats at that time was overwriting the base index.html file. This is a change specific to DietPi and allows other uses of the webserver.

```
and the version of gravity.sh is an old one
```
I will need to test the new gravity.sh and update the DietPi code my end as it may require some changes to fit the DietPi installation. I will contact Jake once this is done.

```
it really doesn't need an entire LASP stack? 
```
This is a DietPi specific change to allow for compatibility with other web based software installation options.

```
It would be great if we could somehow keep DietPi and Pi-hole separate, so we can each work on our projects independently.
```
Yep. As above, the DietPi installation of PiHole will always have a few differences to the installation method and configuration.

**Comment by dschaper** [comment]
Okay, hope I didn't overstep my bounds or anything, wasn't my intention...

**Comment by Fourdee** [comment]
@dschaper No issues my end. But Jake is right, the DietPi installation is different. 
Regardless, if you need to know anything about the installation/configuration or automation code that DietPi uses for PiHole, just let me know.

**Comment by dschaper** [comment]
Thanks, I appreciate it!

**Comment by dschaper** [comment]
@Fourdee Hey, can you fire off another Twitter PM when you get a chance? Just a few questions to follow up with the other issue we were working.

**Comment by Fourdee** [comment]
@dschaper 
> Hey, can you fire off another Twitter PM when you get a chance? Just a few questions to follow up with the other issue we were working.

Not sure I follow, are you @GanymeadeS?

I've been busy with DietPi v123 release (https://github.com/Fourdee/DietPi/pull/428), not had a chance to pick this back up yet. Hopefully start of next week i'll get some time.

**Comment by dschaper** [comment]
@Fourdee Yeah, that's my personal company account. Not a big rush, question about the nullroute stuff.

**Comment by dschaper** [comment]
Closing as curently Out of Scope.
